luigi/contrib/beam_dataflow.py

```diff
@@ -0,0 +1,451 @@
+        if self.num_workers:
+            output.append(f(self.dataflow_params.num_workers, self.num_workers))
+        if self.autoscaling_algorithm:
+            output.append(f(self.dataflow_params.autoscaling_algorithm, self.autoscaling_algorithm))
+        if self.max_num_workers:
+            output.append(f(self.dataflow_params.max_num_workers, self.max_num_workers))
+        if self.disk_size_gb:
+            output.append(f(self.dataflow_params.disk_size_gb, self.disk_size_gb))
+        if self.worker_machine_type:
+            output.append(f(self.dataflow_params.worker_machine_type, self.worker_machine_type))
+        if self.worker_disk_type:
+            output.append(f(self.dataflow_params.worker_disk_type, self.worker_disk_type))
+        if self.network:
+            output.append(f(self.dataflow_params.network, self.network))
+        if self.subnetwork:
+            output.append(f(self.dataflow_params.subnetwork, self.subnetwork))
+        if self.job_name:
+            output.append(f(self.dataflow_params.job_name, self.job_name))
+        if self.service_account:
+            output.append(f(self.dataflow_params.service_account, self.service_account))
+        if self.labels:
+            output.append(f(self.dataflow_params.labels, json.dumps(self.labels)))
+
+        return output
+
+    def _format_input_args(self):
+        job_input = self.input()
+        if isinstance(job_input, luigi.Target):
+            job_input = {"input": job_input}
+        elif not isinstance(job_input, dict):
+            raise ValueError("Input (requires()) must be dict type")
+
+        if not isinstance(self.file_pattern(), dict):
+            raise ValueError('file_pattern() must return a dict type')
+
+        input_args = []
+
+        for (name, targets) in job_input.items():
+            uri_targets = luigi.task.flatten(targets)
+            uris = [self._targets_to_uri_getter.get(
+                uri_target.__class__)(uri_target) for uri_target in uri_targets]
+            if isinstance(targets, dict):
+                """
+                If targets is a dict that means it had multiple outputs.
+                Make the input args in that case "<input key>-<task output key>"
+                """
+                names = ["%s-%s" % (name, key) for key in targets.keys()]
+            else:
+                names = [name] * len(uris)
+            for (arg_name, uri) in zip(names, uris):
+                pattern = self.file_pattern().get(name, 'part-*')
+                input_args.append(
+                    "--%s=%s" %
+                    (arg_name, uri.rstrip('/') + '/' + pattern)
+                )
+
+        return input_args
+
+    def _format_output_args(self):
+        job_output = self.output()
+        if isinstance(job_output, luigi.Target):
+            job_output = {"output": job_output}
+        elif not isinstance(job_output, dict):
```

Comment:
`luigi.Task` also supports `Task`, `list` and `tuple` as return types of `output()`. Consider doing the same?

Reply:
I'm not sure if those types make sense for Dataflow job outputs the same way they do for inputs... maybe `tuple` could be converted to a `dict` but IMO `Task`/`list` are too ambiguous to infer how the user wants them represented as Dataflow job args. what do you think?

Reply:
I not very sure of the use case. But if you believe that is more reasonable, we need to document somehow because that is rather specialized than a normal Luigi `output()`.

Reply:
yeah, I think it's overall going to be a bit awkward because the `output()` is used both in Luigi itself to check `Task` completion, as well as in the Dataflow java/python job (ie in Scio user code, a user could save to two output locations passed in as args, like  `data.saveAsTextFile(args("textOutput")); data.saveAsBigquery(args("bqOutput"))`). I did add some documentation and a unit test to verify that `Task.complete()` works as expected for dict type outputs.